src/Gitonomy/Git/Parser/DiffParser.php

```diff
@@ -22,14 +22,31 @@ class DiffParser extends ParserBase
     protected function doParse()
     {
         $this->files = [];
+        $indexes = [];
+        // Diff contains raw information
+        if (str_starts_with($this->content, ':')) {
+            while ($this->expects(':')) {
+                $this->consumeRegexp('/\d{6} \d{6} /');
+                $oldIndex = $this->consumeShortHash();
+                $this->consume(' ');
+                $newIndex = $this->consumeShortHash();
+                $this->consumeTo("\n");
+                $this->consumeNewLine();
+                $indexes[] = [$oldIndex, $newIndex];
+            }
+            $this->consumeNewLine();
+        } elseif (!$this->isFinished()) {
+            trigger_error('Using Diff::parse without raw information is deprecated. See https://github.com/gitonomy/gitlib/issues/227.', E_USER_DEPRECATED);
```

Comment:
Hello there @Patrick-Beuks @lyrixx,

I'm just wondering:

Would this actually have to result in a deprecation warning?
For other tools using the DiffParser (like grumphp), we don't really care much about the raw index data.

See https://github.com/phpro/grumphp/pull/1199

Reply:
There are edge cases for parsing the diff that would result in an incorrect parse, not finding changed files, see issue #227. 
I will check the use case as you described in your ticket to see if you would have hit the same edge cases tomorrow.